zebra-chain/src/serialization/date_time.rs

```diff
@@ -78,6 +78,38 @@ impl DateTime32 {
     pub fn saturating_elapsed(&self) -> Duration32 {
         DateTime32::now().saturating_duration_since(*self)
     }
+
+    /// Returns the time that is `duration` seconds after this time.
+    /// If the calculation overflows, returns `None`.
+    pub fn checked_add(&self, duration: Duration32) -> Option<DateTime32> {
+        self.timestamp
+            .checked_add(duration.seconds)
+            .map(|timestamp| DateTime32 { timestamp })
```

Comment:
I'm not sure if it's better, but it could be reduced to

```suggestion
            .map(DateTime32::from)
```

Reply:
Fixed for all the methods in 015e090b